sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -1263,8 +1264,16 @@ def get_when_task(input_task_when, depended_conditions):
         task_spec["taskPodTemplate"]["affinity"] = convert_k8s_obj_to_json(op.affinity)
       if op.tolerations:
         task_spec["taskPodTemplate"]['tolerations'] = op.tolerations
+      # process pipeline level first
+      if pipeline_conf and hasattr(pipeline_conf, 'default_pod_node_selector') \
+          and len(pipeline_conf.default_pod_node_selector) > 0:
+        task_spec["taskPodTemplate"]['nodeSelector'] = copy.deepcopy(pipeline_conf.default_pod_node_selector)
```

Comment:
thanks for your rapid response.
but im afraid if i missleading the point.
i think pipelineconf is designed to transfer pipeline level settings, it should be compile as spec.podTemplate in my opinion (which set_image_pull_secrets does so)

Reply:
I see. the end results for the running pods would be the same. but using spec.podTemplate would reduce the CR size. Please re-open the issue and I can have another PR to fix it. Thanks